lib/impl/AzureBlobFiles.js

```diff
@@ -355,10 +365,43 @@ class AzureBlobFiles extends Files {
     }
 
     const presignOptions = Object.assign({ blobName: filePath }, options)
-    const cred = await utils.wrapTVMRequest(this.tvm.getAzureBlobPresignCredentials(presignOptions))
+
+    let cred
+    if (this.hasOwnCredentials) {
+      // generate signature based on options and using own credentials
+      cred = await this._getAzureBlobPresignCredentials(presignOptions)
+    } else {
+      cred = await utils.wrapTVMRequest(this.tvm.getAzureBlobPresignCredentials(presignOptions))
+    }
+
     return this._getUrl(filePath) + '?' + cred.signature
   }
 
+  /**
+   * @memberof AzureBlobFiles
+   * @private
+   */
+  async _getAzureBlobPresignCredentials (params) {
+    if (this._azure.sasCreds) { logAndThrow(new codes.ERROR_UNSUPPORTED_OPERATION({ messageValues: ['generatePresignURL'], sdkDetails: params })) }
```

Comment:
should we make the `messageValues` a bit more precise :  `['generatePresignURL are not supported with Azure Container SAS credentials, please initialize the SDK with Azure storage account credentials instead']`
?